overlord/hookstate/ctlcmd/get_test.go

```diff
@@ -989,3 +993,40 @@ func (s *confdbSuite) TestConfdbGetPreviousInvalid(c *C) {
 	c.Check(stdout, IsNil)
 	c.Check(stderr, IsNil)
 }
+
+func (s *confdbSuite) TestConfdbAccessUnconnectedPlug(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+	tx, err := confdbstate.NewTransaction(s.state, s.devAccID, "network")
+	c.Assert(err, IsNil)
+
+	err = tx.Set(parsePath(c, "wifi.ssid"), "foo")
+	c.Assert(err, IsNil)
+	restore := ctlcmd.MockConfdbstateTransactionForGet(func(*hookstate.Context, *confdb.View, []string) (*confdbstate.Transaction, error) {
+		c.Fatal("should not allow access to confdb")
+		return tx, nil
+	})
+	defer restore()
+
+	for _, plugName := range []string{"read-wifi", "write-wifi", "other"} {
+		err := s.repo.Disconnect("test-snap", plugName, "core", "confdb-slot")
+		c.Assert(err, IsNil)
+	}
+
+	s.state.Unlock()
+	defer s.state.Lock()
+	stdout, stderr, err := ctlcmd.Run(s.mockContext, []string{"get", "--view", ":read-wifi"}, 0)
+	c.Assert(err, ErrorMatches, "cannot access confdb through unconnected plug :read-wifi")
+	c.Check(stdout, IsNil)
+	c.Check(stderr, IsNil)
+
+	stdout, stderr, err = ctlcmd.Run(s.mockContext, []string{"set", "--view", ":write-wifi", "ssid=my-ssid"}, 0)
+	c.Assert(err, ErrorMatches, "cannot access confdb through unconnected plug :write-wifi")
+	c.Check(stdout, IsNil)
+	c.Check(stderr, IsNil)
+
+	stdout, stderr, err = ctlcmd.Run(s.mockContext, []string{"unset", "--view", ":write-wifi", "ssid"}, 0)
+	c.Assert(err, ErrorMatches, "cannot access confdb through unconnected plug :write-wifi")
+	c.Check(stdout, IsNil)
+	c.Check(stderr, IsNil)
```

Comment:
nitpick: should these go to the set_test.go , unset_test.go files?

Reply:
I didn't want to triple the boilerplate so I put them all in the same test